### ISSUE TITLE
[$500] mWeb/Chrome - IOU-Receiver of Send money is shown TBD for a second

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#30431]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.3.91-6

**Reproducible in staging?:** Y

**Reproducible in production?:** Y

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause-Internal Team

**Slack conversation:** @

## Action Performed:
1. Go to https://staging.new.expensify.com/
2. Tap on a 1:1 report of user A and User B in 2 devices
3. Tap plus icon to select Send money as User A
4. Enter an amount
5. Tap next
6. Tap Pay elsewhere
7. Note in user B 1:1 page, TBD displayed for a second and them amount with green tick mark displayed

## Expected Result:
When user A sends money to user B and paid elsewhere, in User B conversation page, TBD must not be shown before amount paid with green tick mark displayed.

## Actual Result:
When user A sends money to user B and paid elsewhere, in User B conversation page, TBD appears for an second before amount paid with green tick mark displayed.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/115492554/775b8b53-5b2a-47a6-9fb4-86aee6117b94

https://github.com/Expensify/App/assets/115492554/f1898357-9be4-40ad-8525-922fe087b315

</details>

<details>
<summary>iOS: Native</summary>

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01b75010b5fae71d08</li>
        <li>Upwork Job ID: 1717511701685063680</li>
        <li>Last Price Increase: 2024-01-17</li>
    </ul>
</details>

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When user receives money, the report preview displays "TBD" for a moment

### What is the root cause of that problem?
The issue is that we are checking whether to show the "TBD" here:

https://github.com/Expensify/App/blob/a2bdeeac7247e3799e06674b9f560057776d0223/src/components/ReportActionItem/ReportPreview.js#L126

The logic to determine this variable is the following:

https://github.com/Expensify/App/blob/950f37a31154aef422fadc194c3791cd84354675/src/libs/ReportUtils.js#L669-L678

It returns `true` if the transactions list is empty.

### What changes do you think we should make in order to solve the problem?

We should modify the `ReportUtils.hasOnlyDistanceRequestTransactions` function to return false if there are no transactions in the report:

```ts
function hasOnlyDistanceRequestTransactions(iouReportID) {
    const allTransactions = TransactionUtils.getAllReportTransactions(iouReportID);
    return !!allTransactions.length && _.all(allTransactions, (transaction) => TransactionUtils.isDistanceRequest(transaction));
}
```

### What alternative solutions did you explore? (Optional)

**Comment by narefyev91** [comment]
@paultsimura does fix for onyx correct order will automatically fix current issue?

**Comment by paultsimura** [comment]
> @paultsimura does correct order fix for onyx will automatically fix current issue?

It should, because the `iouReport` will be present at the moment the `reportAction` is added, and the flickering currently happens because the `iouReport` is missing.

However, that Onyx issue has been open for a while, and I'm curious if we'll have any progress on it anytime soon...

**Comment by narefyev91** [comment]
@mallenexpensify based on the above comment - do you think we should put this on hold until onyx issue will be fixed? I think trying to fix this on FE side may lead to some other issues in future. My suggestion will be to hold on - in case issue is not really influence on prod users. Wdyt

**Comment by mallenexpensify** [comment]
> do you think we should put this on hold until onyx issue will be fixed?

@narefyev91 I have noooooo idea, my knowledge of onyx is extremely limited.  I've added an internal engineer to get feedback.   @hayata-suenaga , you can unassign yourself after answering.   Thx

**Comment by mallenexpensify** [comment]
Bumped to `Weekly` cuz we're on hold pending
- https://github.com/Expensify/App/issues/30219

**Comment by hayata-suenaga** [comment]
on hold

**Comment by hayata-suenaga** [comment]
[Here](https://github.com/Expensify/App/issues/30219#issuecomment-1819068342) is the latest update for https://github.com/Expensify/App/issues/30219

**Comment by hayata-suenaga** [comment]
https://github.com/Expensify/App/issues/30219 is still accepting proposals.

**Comment by hayata-suenaga** [comment]
investigation for https://github.com/Expensify/App/issues/30219 is still going on

**Comment by hayata-suenaga** [comment]
[The PR](https://github.com/Expensify/App/pull/32543) for the issue #30219 is ready for review

**Comment by mallenexpensify** [comment]
Commented on the PR
- https://github.com/Expensify/App/pull/32543#issuecomment-1874766887

**Comment by mallenexpensify** [comment]
Off hold, https://github.com/Expensify/App/issues/30219 hit production

@izarutskaya can you retest? I don't have an android phone handy to test with. Thx

**Comment by hayata-suenaga** [comment]
I asked in the QA channel to get somebody test this

**Comment by kbecciv** [comment]
Issue is fixed, build 1.4.27.0

https://github.com/Expensify/App/assets/93399543/b0563ac6-53d8-4306-bdfa-965df3c43b50

**Comment by hayata-suenaga** [comment]
since this was fixed, I'm closing this issue
